plugins/Rockets/SnapshotTask.cpp

```diff
@@ -0,0 +1,188 @@
+        _camera = engine.createCamera();
+        _scene =
+            engine.createScene(*_params.animParams, *_params.geometryParams,
+                               *_params.volumeParams);
+
+        _renderer = engine.createRenderer(*_params.animParams,
+                                          *_params.renderingParams);
+
+        const auto &renderer = engine.getRenderer();
+        _renderer->setCurrentType(renderer.getCurrentType());
+        _renderer->clonePropertiesFrom(renderer);
+        if (_params.camera)
+        {
+            *_camera = *_params.camera;
+            _camera->setCurrentType(engine.getCamera().getCurrentType());
+            _camera->clonePropertiesFrom(engine.getCamera());
+        }
+        else
+            *_camera = engine.getCamera();
+
+        _scene->copyFrom(engine.getScene());
+    }
+    else
+    {
+        _camera = engine.getCameraPtr();
+        _renderer = engine.getRendererPtr();
+    }
+}
+
+ImageGenerator::ImageBase64 SnapshotFunctor::_renderSnapshot(
+    const size_t numAccumulations)
+{
+    const auto nameShort =
+        _params.name.empty() ? "" : " " + shortenString(_params.name);
+    const auto msg = "Render snapshot" + nameShort + "...";
+
+    const auto isStereo =
+        _camera->hasProperty("stereo") && _camera->getProperty<bool>("stereo");
+    const auto names = isStereo ? strings{"0L", "0R"} : strings{"default"};
+
+    std::vector<FrameBufferPtr> frameBuffers;
+    for (const auto &name : names)
+    {
+        frameBuffers.push_back(
+            _engine.createFrameBuffer(name, _params.size,
+                                      FrameBufferFormat::rgba_i8));
+    }
+
+    for (size_t i = 0; i < numAccumulations; i++)
+    {
+        for (auto &frameBuffer : frameBuffers)
+        {
+            _camera->setBufferTarget(frameBuffer->getName());
+            _camera->markModified(false);
+            _camera->commit();
+            _camera->resetModified();
+            _renderer->render(frameBuffer);
+            frameBuffer->incrementAccumFrames();
+        }
+
+        progress(msg, 1.f / numAccumulations,
+                 static_cast<float>(i) / numAccumulations);
+    }
+
+    return _imageGenerator.createImage(frameBuffers, _params.format,
+                                       _params.quality);
+}
+
+ImageGenerator::ImageBase64 SnapshotFunctor::_takeSnapshotConcurrent()
+{
+    _scene->commit();
+
+    _camera->updateProperty("aspect", double(_params.size.x) / _params.size.y);
+    _camera->commit();
+
+    if (_params.renderingParams)
+    {
+        _params.renderingParams->setSamplesPerPixel(1);
+        _params.renderingParams->setSubsampling(1);
+    }
+
+    _scene->commitLights();
+
+    _renderer->setCamera(_camera);
+    _renderer->setScene(_scene);
+    _renderer->commit();
+
+    return _renderSnapshot(_params.samplesPerPixel);
+}
+
+ImageGenerator::ImageBase64 SnapshotFunctor::_takeSnapshotSerial()
+{
+    auto lock{_engine.getRenderScopeLock()};
```

Comment:
I don't see where you are setting the snapshot size in this case. In the concurrent snapshot the rendering parameters (with the buffer size) are passed to the constructor of the new renderer, but here you use the same renderer.

Reply:
It is set on line 112 when creating the buffer.